sdk/keyvault/keyvault-secrets/package.json

```diff
@@ -19,51 +19,44 @@
   "bugs": {
     "url": "https://github.com/Azure/azure-sdk-for-js/issues"
   },
-  "main": "./dist/index.js",
-  "module": "dist-esm/src/index.js",
-  "types": "./types/keyvault-secrets.d.ts",
+  "main": "./dist/commonjs/index.js",
+  "module": "./dist/esm/index.js",
+  "types": "./dist/commonjs/index.d.ts",
```

Comment:
I don't recall what we are doing for the other packages, but is it intentional that we're not using the rolled-up d.ts from API extractor (dist/keyvault-secrets.d.ts) here? If so, maybe we don't need to ship the rolled-up version.

Reply:
Hm good Q. It is intentional as far as I understand it but @mpodwysocki can confirm. IIUC tshy now manages creating dts files per kind (CJS, ESM, etc) and while we no longer get a single nicely rolled up dts I don't think it's a huge loss (or a loss at all).

So maybe we disable the rollup dts option in API Extractor for migrated packages? Otherwise we're just shipping `dist/keyvault-secrets.d.ts` unnecessarily...

Reply:
Do we know if some of our tooling need the rolled up types, e.g. docs generation?

Reply:
One thing that's useful about it is that it generates warnings when you forget to document public surface. But I'm pretty sure that those warnings show up in the APIView and/or review .md file anyway.

If we do have tooling that requires it, I wonder if that tooling depends on the rolled up types being shipped or not. If not, maybe we could just make it output somewhere that's not under `dist/`.

Reply:
AFAIK our tooling all hinges on the docModel output by API Extractor under `temp`:

- keyvault-keys.api.json gets uploaded to APIView
- keyvault-keys.api.md gets moved to review/ 

The rolled up dts file, AFAIK, is not used by our tooling - but please feel free to fact check me because I don't know for sure!

Reply:
Yes, the rolled up (or public trimmed as api-extractor calls it) dts files is not used any more after migration.  There's one use case though: if you have @beta tag on public API, api-extractor removes them from the public trimmed file, but will keep it in beta-trimmed file.   Sounds that additional efforts would be needed for this scenario after migrating to tshy

Reply:
Good discussion - probably outside the scope of blocking this PR so I will resolve the conversation (per GH merge rules) but we should definitely continue to think about this. I created https://github.com/Azure/azure-sdk-for-js/issues/31310 to track that discussion